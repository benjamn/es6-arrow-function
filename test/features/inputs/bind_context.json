{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "CallExpression",
        "callee": {
          "type": "MemberExpression",
          "computed": false,
          "object": {
            "type": "Identifier",
            "name": "stream"
          },
          "property": {
            "type": "Identifier",
            "name": "on"
          }
        },
        "arguments": [
          {
            "type": "Literal",
            "value": "data",
            "raw": "\"data\""
          },
          {
            "type": "ArrowFunctionExpression",
            "id": null,
            "params": [
              {
                "type": "Identifier",
                "name": "d"
              }
            ],
            "defaults": [],
            "body": {
              "type": "AssignmentExpression",
              "operator": "+=",
              "left": {
                "type": "MemberExpression",
                "computed": false,
                "object": {
                  "type": "ThisExpression"
                },
                "property": {
                  "type": "Identifier",
                  "name": "data"
                }
              },
              "right": {
                "type": "Identifier",
                "name": "d"
              }
            },
            "rest": null,
            "generator": false,
            "expression": true
          }
        ]
      }
    }
  ]
}
