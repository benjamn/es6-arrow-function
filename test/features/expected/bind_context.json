{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "CallExpression",
        "callee": {
          "type": "MemberExpression",
          "computed": false,
          "object": {
            "type": "Identifier",
            "name": "stream"
          },
          "property": {
            "type": "Identifier",
            "name": "on"
          }
        },
        "arguments": [
          {
            "type": "Literal",
            "value": "data",
            "raw": "\"data\""
          },
          {
            "type": "CallExpression",
            "callee": {
              "type": "MemberExpression",
              "computed": false,
              "object": {
                "type": "FunctionExpression",
                "id": null,
                "params": [
                  {
                    "type": "Identifier",
                    "name": "d"
                  }
                ],
                "defaults": [],
                "body": {
                  "type": "BlockStatement",
                  "body": [
                    {
                      "type": "ReturnStatement",
                      "argument": {
                        "type": "AssignmentExpression",
                        "operator": "+=",
                        "left": {
                          "type": "MemberExpression",
                          "computed": false,
                          "object": {
                            "type": "ThisExpression"
                          },
                          "property": {
                            "type": "Identifier",
                            "name": "data"
                          }
                        },
                        "right": {
                          "type": "Identifier",
                          "name": "d"
                        }
                      }
                    }
                  ]
                },
                "rest": null,
                "generator": false,
                "expression": false
              },
              "property": {
                "type": "Identifier",
                "name": "bind"
              }
            },
            "arguments": [
              {
                "type": "ThisExpression"
              }
            ]
          }
        ]
      }
    }
  ]
}
